### ISSUE TITLE
kAudioObjectPropertyElementMaster has been replaced with kAudioObjectPropertyElementMain in MacOS

### ISSUE THREAD

**Issue by tcaputi** [obsproject/obs-studio#9671]
### Operating System Info

macOS 13

### OBS Studio Version

Git

### OBS Studio Log URL

none

### Expected Behavior

The project should build on MacOS 13

### Current Behavior

The project doesn't compile on MacOS 13

### Steps to Reproduce

1. Fork / clone https://github.com/obsproject/obs-plugintemplate
2. Install dependencies
3. Run `cmake`

### Anything else we should know?

I am working on a plugin, which downloaded 29.1.2 as a dependency. When I tried to run cmake i got an error. after doing a little investigating I found that the problem is that MacOS has depricated kAudioObjectPropertyElementMaster, renaming it to kAudioObjectPropertyElementMain. Simply changing the name fixes the error. A proper fix for this will probably require conditional compilation to select the correct name. 

Here is the relevant code from AudioHardwareBase.h (from apple's sdk)

```
/*!
    @enum           Property Address Constants
    @abstract       The valid values for the scope in a property address.
    @constant       kAudioObjectPropertyScopeGlobal
                        The AudioObjectPropertyScope for properties that apply to the object as a
                        whole. All objects have a global scope and for most it is their only scope.
    @constant       kAudioObjectPropertyScopeInput
                        The AudioObjectPropertyScope for properties that apply to the input side of
                        an object.
    @constant       kAudioObjectPropertyScopeOutput
                        The AudioObjectPropertyScope for properties that apply to the output side of
                        an object.
    @constant       kAudioObjectPropertyScopePlayThrough
                        The AudioObjectPropertyScope for properties that apply to the play through
                        side of an object.
    @constant       kAudioObjectPropertyElementMain
                        The AudioObjectPropertyElement value for properties that apply to the main
                        element or to the entire scope.
    @constant       kAudioObjectPropertyElementMaster
						The deprecated synonym for kAudioObjectPropertyElementMain
*/
CF_ENUM(AudioObjectPropertyScope)
{
    kAudioObjectPropertyScopeGlobal         = 'glob',
    kAudioObjectPropertyScopeInput          = 'inpt',
    kAudioObjectPropertyScopeOutput         = 'outp',
    kAudioObjectPropertyScopePlayThrough    = 'ptru',
    kAudioObjectPropertyElementMain			= 0,
    kAudioObjectPropertyElementMaster API_DEPRECATED_WITH_REPLACEMENT("kAudioObjectPropertyElementMain", macos(10.0, 12.0), ios(2.0, 15.0), watchos(1.0, 8.0), tvos(9.0, 15.0))	= kAudioObjectPropertyElementMain
};
```

**Comment by gxalpha** [comment]
Could you post the full commands you’re running, and a bigger excerpt of the error message? Also, what do you mean by "Install dependencies"? Cmake should auto-install the dependencies needed.
I’m aware of this warning, but many people (including me) are compiling OBS just fine on macOS 13 and 14.

**Comment by tcaputi** [comment]
So.... I was just told by someone on the discord about the build scripts in the github folder. I was doing what I always usually do which is just:

```
mkdir build
cd build
cmake ..
```


The output produced by just running the cmake command wasn't super helpful:
```
-- Setting up Pre-Built obs-deps (universal)
-- Setting up Pre-Built obs-deps (universal) - skipped
-- Setting up Pre-Built Qt6 (universal)
-- Setting up Pre-Built Qt6 (universal) - skipped
-- Setting up OBS sources (universal)
-- Setting up OBS sources (universal) - done
-- Configure OBS sources (universal)
                      _                   _             _ _       
                 ___ | |__  ___       ___| |_ _   _  __| (_) ___  
                / _ \| '_ \/ __|_____/ __| __| | | |/ _` | |/ _ \ 
               | (_) | |_) \__ \_____\__ \ |_| |_| | (_| | | (_) |
                \___/|_.__/|___/     |___/\__|\__,_|\__,_|_|\___/ 

OBS:  Application Version: 29.1.2 - Build Number: 3
==================================================================================


------------------------       Enabled Features           ------------------------
 - OpenGL renderer
------------------------       Disabled Features          ------------------------
 - Plugin Support
 - User Interface
----------------------------------------------------------------------------------
-- Configure OBS sources (universal) - done
-- Build OBS sources (universal)
** BUILD FAILED **


The following build commands failed:
        CompileC /Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/build_universal/build/libobs.build/Debug/Objects-normal/x86_64/coreaudio-enum-devices.o /Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/libobs/audio-monitoring/osx/coreaudio-enum-devices.c normal x86_64 c com.apple.compilers.llvm.clang.1_0.compiler (in target 'libobs' from project 'obs-studio')
(1 failure)
CMake Error at cmake/common/buildspec_common.cmake:89 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 65"

Call Stack (most recent call first):
  cmake/common/buildspec_common.cmake:214 (_setup_obs_studio)
  cmake/macos/buildspec.cmake:23 (_check_dependencies)
  cmake/macos/buildspec.cmake:34 (_check_dependencies_macos)
  cmake/macos/defaults.cmake:23 (include)
  CMakeLists.txt:11 (include)

```


So I edited `buildspec_common.cmake` to remove all the `OUTPUT_QUIET`s. Then I was able to see this error (truncated cause there is a LOT of output)
```
CompileC /Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/build_universal/build/libobs.build/Debug/Objects-normal/x86_64/coreaudio-enum-devices.o /Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/libobs/audio-monitoring/osx/coreaudio-enum-devices.c normal x86_64 c com.apple.compilers.llvm.clang.1_0.compiler (in target 'libobs' from project 'obs-studio')
    cd /Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c -ivfsstatcache /var/folders/dz/1kp_39wj3xl_10zl4b1w8gz40000gn/C/com.apple.DeveloperTools/15.0-15A240d/Xcode/SDKStatCaches.noindex/macosx14.0-23A334-4a76ee93ef69e462914ffe9883de22b8.sdkstatcache -target x86_64-apple-macos13.5 -fmessage-length\=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit\=0 -fno-color-diagnostics -std\=c17 -Wno-trigraphs -fpascal-strings -O0 -Werror -Wno-missing-field-initializers -Wno-missing-prototypes -Werror\=return-type -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wunused-parameter -Wunused-variable -Wunused-value -Wempty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wfour-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wsign-compare -Wshorten-64-to-32 -Wpointer-sign -Wnewline-eof -Wno-implicit-fallthrough -DCMAKE_INTDIR\=\"Debug\" -Dlibobs_EXPORTS -DIS_LIBOBS -DSIMDE_ENABLE_OPENMP -DENABLE_HEVC -DHAVE_OBSCONFIG_H -DDEBUG -D_DEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk -fasm-blocks -fno-strict-aliasing -Wdeprecated-declarations -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-comma -Werror\=block-capture-autoreleasing -Wno-strict-prototypes -Wno-semicolon-before-method-body -I/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/build_universal/libobs/Debug/include -I/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/build_universal/libobs -I/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/libobs -I/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/build_universal/config -I/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/deps/libcaption -I/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/deps/uthash -isystem /Users/tom/Development/projects/autovod-plugin/.deps/obs-deps-2023-04-12-universal/include -I/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/build_universal/build/libobs.build/Debug/DerivedSources-normal/x86_64 -I/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/build_universal/build/libobs.build/Debug/DerivedSources/x86_64 -I/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/build_universal/build/libobs.build/Debug/DerivedSources -Wvla -Wformat-security -Wno-error\=shorten-64-to-32 -F/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/build_universal/libobs/Debug -F/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks -F/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks -std\=c17 -fPIC -fvisibility\=hidden -fopenmp-simd -MMD -MT dependencies -MF /Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/build_universal/build/libobs.build/Debug/Objects-normal/x86_64/coreaudio-enum-devices.d --serialize-diagnostics /Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/build_universal/build/libobs.build/Debug/Objects-normal/x86_64/coreaudio-enum-devices.dia -c /Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/libobs/audio-monitoring/osx/coreaudio-enum-devices.c -o /Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/build_universal/build/libobs.build/Debug/Objects-normal/x86_64/coreaudio-enum-devices.o
/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/libobs/audio-monitoring/osx/coreaudio-enum-devices.c:24:9: error: 'kAudioObjectPropertyElementMaster' is deprecated: first deprecated in macOS 12.0 [-Werror,-Wdeprecated-declarations]
                                           kAudioObjectPropertyElementMaster};
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                           kAudioObjectPropertyElementMain
In file included from /Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/libobs/audio-monitoring/osx/coreaudio-enum-devices.c:2:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/CoreAudio.framework/Headers/CoreAudio.h:18:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardware.h:85:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareBase.h:208:5: note: 'kAudioObjectPropertyElementMaster' has been explicitly marked deprecated here
    kAudioObjectPropertyElementMaster API_DEPRECATED_WITH_REPLACEMENT("kAudioObjectPropertyElementMain", macos(10.0, 12.0), ios(2.0, 15.0), watchos(1.0, 8.0), tvos(9.0, 15.0)) = kAudioObjectPropertyElementMain
    ^
/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/libobs/audio-monitoring/osx/coreaudio-enum-devices.c:72:9: error: 'kAudioObjectPropertyElementMaster' is deprecated: first deprecated in macOS 12.0 [-Werror,-Wdeprecated-declarations]
                                           kAudioObjectPropertyElementMaster};
                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                           kAudioObjectPropertyElementMain
In file included from /Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/libobs/audio-monitoring/osx/coreaudio-enum-devices.c:2:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/CoreAudio.framework/Headers/CoreAudio.h:18:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardware.h:85:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareBase.h:208:5: note: 'kAudioObjectPropertyElementMaster' has been explicitly marked deprecated here
    kAudioObjectPropertyElementMaster API_DEPRECATED_WITH_REPLACEMENT("kAudioObjectPropertyElementMain", macos(10.0, 12.0), ios(2.0, 15.0), watchos(1.0, 8.0), tvos(9.0, 15.0)) = kAudioObjectPropertyElementMain
    ^
/Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/libobs/audio-monitoring/osx/coreaudio-enum-devices.c:119:3: error: 'kAudioObjectPropertyElementMaster' is deprecated: first deprecated in macOS 12.0 [-Werror,-Wdeprecated-declarations]
                kAudioObjectPropertyElementMaster};
                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                kAudioObjectPropertyElementMain
In file included from /Users/tom/Development/projects/autovod-plugin/.deps/obs-studio-29.1.2/libobs/audio-monitoring/osx/coreaudio-enum-devices.c:2:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/CoreAudio.framework/Headers/CoreAudio.h:18:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardware.h:85:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks/CoreAudio.framework/Headers/AudioHardwareBase.h:208:5: note: 'kAudioObjectPropertyElementMaster' has been explicitly marked deprecated here
    kAudioObjectPropertyElementMaster API_DEPRECATED_WITH_REPLACEMENT("kAudioObjectPropertyElementMain", macos(10.0, 12.0), ios(2.0, 15.0), watchos(1.0, 8.0), tvos(9.0, 15.0)) = kAudioObjectPropertyElementMain
    ^
3 errors generated.


```

**Comment by Conan-Kudo** [comment]
Does it work when you build the plugin template against OBS v30-beta3 or git master instead?

**Comment by gxalpha** [comment]
> So.... I was just told by someone on the discord about the build scripts in the github folder. I was doing what I always usually do which is just:
> 
> ```
> mkdir build
> cd build
> cmake ..
> ```

Generally I'd recommend to invoke cmake in the top-level directory rather than the build directory (so you'd just do `cmake -S . -B build`, where `-S .` specifies the source directory and `-B build` the build directory; although `-S .` is redundant if it's the current directory).

In this case, the problem seems to be that the deployment target defaults to your macOS version, where this warning is emitted. This would be fixed by using the default cmake preset for macOS, which sets the deployment version to 11.0 (and also includes the `-B` build directory).

Full cmake command (in the source directory) would be `cmake --preset macos`. To build, either open the resulting Xcode project and compile from there, or I believe `cmake --build --preset macos` (also from the source directory) should also do the job.

**Comment by tcaputi** [comment]
Ah, ok. Feel free to close this then or keep it as a reminder for when you guys update the deployment version. Sorry for the noise.

**Comment by tcaputi** [comment]
@Conan-Kudo oh hey!

i can see the offending line of code is still in coreaudio-enum-devices.c so i didnt bother trying to rejigger the build system

**Comment by Conan-Kudo** [comment]
@tcaputi 😂👋

**Comment by gxalpha** [comment]
Forgot to link, but this was addressed by #9717
